#15588 Fixed incorrect image urls in multistore xml sitemap

StevenGuapaBV opened the pull request:
### Description
Added the appEmulation to get the correct image Url for the generated sitemaps. This was needed because the catalog image helper context had incorrect Url data. 

### Fixed Issues (if relevant)
1. magento/magento2#15588: Images in XML sitemap are always linked to base store in multistore#15588

### Manual testing scenarios
1. In a multistore environment generate multiple sitemaps. Check if the <image:loc> has the correct base URL for set store. 

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

StevenGuapaBV committed:
#15588 

Added the appEmulation to get the correct image Url for the generated sitemaps. Context had incorrect Url data.

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=15689) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=15689) before we can accept your contribution.<br/>

StevenGuapaBV committed:
#15588 Refactor for code neatness Travis CI.

 48 | ERROR | [x] Opening parenthesis of a multi-line function call must be the last content on the line

StevenGuapaBV committed:
Update for Codacy removed whitespace

StevenGuapaBV committed:
#15588 Fixed incorrect image urls sitemap

Change for travisCI

magento-engcom-team commented:
@StevenGuapaBV thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

dmanners commented on app/code/Magento/Sitemap/Controller/Adminhtml/Sitemap/Generate.php:
```diff
@@ -23,7 +44,14 @@ public function execute()
+                    true
+                );
                 $sitemap->generateXml();
+                $this->appEmulation->stopEnvironmentEmulation();
```
As mentioned by @JeroenVanLeusden I think it would be a good idea to make sure we are stopping this emulation in all cases including exception cases. https://github.com/magento/magento2/pull/15689#issuecomment-394291002

StevenGuapaBV committed:
#15588 Fixed incorrect image urls in multistore xml sitemap

placed the stop Emulation in a finally. Now it will always be executed.

StevenGuapaBV commented:
Hi @dmanners  and @JeroenVanLeusden. I changed the logic and put the stop in a finally. Is that okay as well? Seems to make more sense than a try catch in a try catch block. 

A finally will always be executed and the stop emulation does not throw an exception of its own.

dmanners committed:
magento/magento2#15588: Remove extra line from if

dmanners committed:
magento/magenot2#15588: add dependency in a backwards compatible way

magento-engcom-team commented:
Hi @StevenGuapaBV. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.2.6 release.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

StevenGuapaBV commented:
Hi @magento-engcom-team,

I made the forwardport for the 2.3 release line via the porting tool. This should be done now.
